tests/pipeline/test_pipeline.py

```diff
@@ -163,6 +163,51 @@ def test_file_format_resolution() -> None:
         pipeline.run([1, 2, 3], table_name="numbers", loader_file_format="insert_values")
 
 
+def test_staging_cleared() -> None:
+    pipeline = dlt.pipeline(
+        pipeline_name="test_staging_cleared",
+        destination="bigquery",
+        staging="filesystem",
+        full_refresh=True,
+    )
+    info = pipeline.run(
+        [{"field": 1}, {"field": 2}, {"field": 3}],
+        table_name="staging_cleared_resource",
+    )
+    assert_load_info(info)
+
+    _, staging = pipeline._get_destination_clients(pipeline.default_schema)
+    assert staging.list_table_files("staging_cleared_resource") == []
```

Comment:
Checking that there are no files

Reply:
@IlyaFaer i think it would be better to have a parametrized test that runs the same code once with the prop set to true and once to false and so the asserts accordingly at the end